### DISCUSSION TITLE
EBSD analysis reference to Documentation in MTEX

### DISCUSSION THREAD

**Discussion by wuwuhd** [ORTools4MTEX/ORTools#58] [Q&A] [answered]
Hi, 
When I was dealing with EBSD data in a dual phase (mixtures of fcc asutenite and bcc martensite) steel, I always encounter a problem that the command  is only permitted for a single phase! For example, when I calculated the Geometrically Necessary Dislocations, the kind of thing was sent when the first step code run.  I refer to the code here: https://mtex-toolbox.github.io/GND.html
And my codes are: 
%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ctf',...
  'convertSpatial2EulerReferenceFrame');
% define the color key
ipfKey = ipfHSVKey(ebsd);
ipfKey.inversePoleFigureDirection = yvector;

% and plot the orientation data
plot(ebsd,ipfKey.orientation2color(ebsd.orientations),'micronBar','off','figSize','medium')

![image](https://github.com/ORTools4MTEX/ORTools/assets/149949577/74486cba-67ff-462a-a63b-89136befdae5)

Best wish!
wu

**Comment by AzdiarGazder** [comment, marked as answer]
Hi @wuwuhd,

I understand where you're coming from but there is sound rationale for doing so.
- MTEX was created to work for all users with different kinds of maps from different vendors, convention systems, map grid types, and phases. So the easiest way to script for a very large audience with different datasets is to modularise the functions so that it works for everyone and not just for a few specific cases.

- Many functions inherently deal with a single phase because they are quantifying phenomena specific to a single crystal system. Consequently, the MTEX documentation shows examples on how to work with these functions using single phase maps (for the most part). 

   For example, in your last question on [the distribution of grain boundary normals]( https://mtex-toolbox.github.io/BoundaryNormalDistribution.html), segmenting the grain boundary population into < 57 deg and > 57 deg subsets only makes sense for the fcc phase. This is because fcc-fcc boundaries return a predominant twin system at 60 deg/<111>.  Consequently, it does not make sense doing this segmentation for the bcc case and neither does it make sense for fcc-bcc interphase boundaries.

- When you are working on multi-phase maps, the simplest thing to do is to define each of your various phases beforehand and then call them whenever needed. Yes, it is an extra step, but it makes the script very clear to follow and also ensures no mistakes on your part. Here is an example:

```matlab
% clear variables
clc; clear all; clear hidden; close all;

% start Mtex
startup_mtex;

% define Mtex plotting convention as X = right, Y = up
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','outOfPlane');
setMTEXpref('FontSize', 14);

% import the dataset
ebsd = EBSD.load('Duplex.ctf','interface','ctf',...
    'convertEuler2SpatialReferenceFrame');
ebsd = ebsd('indexed');


%% calculate the grains
% identify grains
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'angle',5*degree);
% remove small clusters
ebsd(grains(grains.grainSize <= 5)) = [];
% re-calculate grains
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'angle',5*degree);
% smooth grains
grains = smooth(grains,5);
% calculate the grain boundaries
gB = grains.boundary;

% define the fcc phase
ebsd_fcc = ebsd('Fe-FCC');
grains_fcc = grains('Fe-FCC');
gB_fcc = gB('Fe-FCC','Fe-FCC');

cond = gB_fcc.misorientation.angle > 57 * degree;

figure
plot(ebsd_fcc,ebsd_fcc.orientations);
hold all;
plot(gB_fcc,'lineWidth',2);
plot(gB_fcc(cond),'lineWidth',2,'lineColor','w');
hold off;

figure
gbnd1 = calcGBPD(gB_fcc(cond),ebsd_fcc);
gbnd2 = calcGBPD(gB_fcc(~cond),ebsd_fcc);

contourf(gbnd1);
mtexTitle('GBPD for misorientation angle \(> 57^{\circ}\)');
mtexColorMap parula;
nextAxis;

contourf(gbnd2);
mtexTitle('GBPD for misorientation angle \(< 57^{\circ}\)');
mtexColorMap parula;
mtexColorbar;
```


Hope this helps.

<ins>**Lastly, and importantly, if an explanation from the maintainers resolves your issue, please press the "Mark as answer" button on the appropriate response. Your help in this regard enables us to close discussions as complete. **</ins>

Warm regards,
Azdi

**Reply by wuwuhd** [reply]
Hi, Azdiar
Thanks for your help, it works for me! But I still have a question about Geometrically Necessary Dislocations, and I will open a new discussion. Looking forward to receiving your reply in the new discussion！
Best wish!
wu
